rclcpp/test/rclcpp/test_publisher.cpp

```diff
@@ -536,3 +537,45 @@ TEST_F(TestPublisher, get_network_flow_endpoints_errors) {
     EXPECT_NO_THROW(publisher->get_network_flow_endpoints());
   }
 }
+
+TEST_F(TestPublisher, check_wait_for_all_acked_return) {
```

Comment:
i see this is unit test for interfaces in rclcpp, but i think the following test cases are `nice to have` to make sure all the call stack works fine.

- actually issue `rcl_publisher_wait_for_all_acked` on publisher w/o `mocking_utils::patch_and_return`.
- QoS setting for BestEffort and Reliable since those behaves differently in implementation.

Reply:
Yes, I should add test to confirm call stack works well.